lib/Dialect/Triton/Transforms/ReorderBroadcast.cpp

```diff
@@ -116,18 +116,21 @@ struct MoveBroadcastAfterElementwisePattern
 
     auto operands = op->getOperands();
     bool seenBroadcast = false;
+    Type srcType;
     for (auto operand : operands) {
       auto definingOp = operand.getDefiningOp();
       if (!definingOp) {
         return mlir::failure();
       }
 
       if (auto broadcastOp = llvm::dyn_cast<triton::BroadcastOp>(definingOp)) {
-        if (seenBroadcast) {
-          // Only support one broadcasted argument for now
+        if (!seenBroadcast) {
+          seenBroadcast = true;
+          srcType = broadcastOp.getSrc().getType();
+        } else if (srcType != broadcastOp.getSrc().getType()) {
+          // If the broadcast have different types we cannot re-order.
```

Comment:
This is a nice improvement. I think you can generalize it slightly further by having a common `srcShape` and allowing the scalar types to differ. This would allow the pattern to work for `AddPtrOp` for example.

Reply:
sorry for the delay, thanks for the advice. I'm sending another PR to handle this.